File: backend-midway/src/controller/data-cleanup.controller.ts
import { Controller, Post, Inject, Config } from '@midwayjs/core';
import { ApiTags, ApiOperation } from '@midwayjs/swagger';
import { DataCleanupService } from '../service/data-cleanup.service';

@Controller('/data-cleanup')
@ApiTags(['data-cleanup'])
export class DataCleanupController {
  @Inject()
  dataCleanupService: DataCleanupService;

  @Config('defaultProjectId')
  defaultProjectId: string;

  @Post('/clear-all')
  @ApiOperation({ summary: '清除所有监控数据' })
  async clearAll() {
    try {
      await this.dataCleanupService.clearAllData(this.defaultProjectId);
      return { success: true, message: '所有监控数据已清除' };
    } catch (error) {
      console.error('Clear all data failed:', error);
      return { success: false, message: '清除数据失败: ' + error.message };
    }
  }
}

File: backend-springboot/src/main/java/com/monitor/controller/DataCleanupController.java
package com.monitor.controller;

import com.monitor.service.DataCleanupService;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

/**
 * 数据清理控制器
 */
@RestController
@RequestMapping("/data-cleanup")
@RequiredArgsConstructor
public class DataCleanupController {
    
    private final DataCleanupService dataCleanupService;
    
    @Value("${default-project-id:001}")
    private String defaultProjectId;
    
    @PostMapping("/clear-all")
    public ResponseEntity<Map<String, Object>> clearAll() {
        try {
            dataCleanupService.clearAllData(defaultProjectId);
            return ResponseEntity.ok(Map.of(
                "success", true,
                "message", "所有监控数据已清除"
            ));
        } catch (Exception e) {
            System.err.println("Clear all data failed: " + e.getMessage());
            return ResponseEntity.ok(Map.of(
                "success", false,
                "message", "清除数据失败: " + (e.getMessage() != null ? e.getMessage() : "未知错误")
            ));
        }
    }
}

